Azure-Samples/azure-search-openai-demo
# Fixed type annotations per mypy and added mypy to CICD

## Purpose

I noticed some incorrect types when running mypy on code I was writing, so I went ahead and fixed all the type annotations, as well as added a CI/CD check for mypy going forward.

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Test still pass
* I will do a test "azd up" to check prepdocs. I don't have ACLs setup for testing yet, so if you have concerns about the manageacls changes, please test those.
